queue.js

```diff
@@ -0,0 +1,74 @@
+const {
+    SQSClient,
+    ReceiveMessageCommand,
+    SendMessageCommand,
+    DeleteMessageCommand,
+} = require('@aws-sdk/client-sqs');
+
+const sqsClient = new SQSClient({
+    region: process.env.AWS_REGION,
+    credentials: {
+        accessKeyId: process.env.AWS_ACCESS_KEY_ID,
+        secretAccessKey: process.env.AWS_SECRET_ACCESS_KEY,
+    },
+});
+
+const queueURL = process.env.SQS_QUEUE_URL;
+const dlqURL = process.env.SQS_DLQ_URL;
+
+const getMessage = async () => {
+    const receiveMessageCommand = new ReceiveMessageCommand({
+        QueueUrl: queueURL,
+        MaxNumberOfMessages: 1,
+        WaitTimeSeconds: 20,
+    });
+    const res = await sqsClient.send(receiveMessageCommand);
+    return res;
+};
+
+const deleteMessage = async (receiptHandle) => {
+    const deleteMessageCommand = new DeleteMessageCommand({
+        QueueUrl: queueURL,
+        ReceiptHandle: receiptHandle,
+    });
+    const res = await sqsClient.send(deleteMessageCommand);
+    return res;
+};
+
+const deleteMessageWithRetries = async (receiptHandle, retries = 0) => {
+    try {
+        await deleteMessage(receiptHandle);
+    } catch (err) {
+        if (retries < 3) {
+            await deleteMessageWithRetries(receiptHandle, retries + 1);
+        } else {
+            throw err;
+        }
+    }
+};
+
+const sendToDLQ = async (message) => {
```

Comment:
this implementation looks very suspicious. i couldn't imagine why one would explicitly add something to DLQ. i thought it is controlled by handler return code

Reply:
Yes I think my comment here explains what I was thinking: https://github.com/alex-808/flashcard-service/pull/1#discussion_r1296594612. Do you think it would be best to just let the queue itself move it after 3 attempts?

Reply:
thx for clarification. yes, this makes sense